# Questão 1

In [4]:
import numpy as np
from sklearn.datasets import load_wine
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score
from mlxtend.evaluate import paired_ttest_5x2cv

In [6]:
X, y = load_wine(return_X_y=True)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [10]:
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

In [11]:
clf1 = LogisticRegression(random_state=42, max_iter=1000)
clf1.fit(X_train_scaled, y_train)
y_pred1 = clf1.predict(X_test_scaled)
acc1 = accuracy_score(y_test, y_pred1)
print(f"Acurácia (Regressão Logística): {acc1:.4f}")

Acurácia (Regressão Logística): 1.0000


In [9]:
clf2 = RandomForestClassifier(random_state=42)
clf2.fit(X_train, y_train)
y_pred2 = clf2.predict(X_test)
acc2 = accuracy_score(y_test, y_pred2)
print(f"Acurácia (Random Forest): {acc2:.4f}")

Acurácia (Random Forest): 1.0000


In [12]:
clf2 = RandomForestClassifier(random_state=42)
clf2.fit(X_train_scaled, y_train)
y_pred2 = clf2.predict(X_test_scaled)
acc2 = accuracy_score(y_test, y_pred2)
print(f"Acurácia (Random Forest): {acc2:.4f}")

Acurácia (Random Forest): 1.0000


In [14]:
X_scaled = scaler.fit_transform(X)

In [15]:
t, p = paired_ttest_5x2cv(estimator1=clf1,
                            estimator2=clf2,
                            X=X_scaled, y=y,
                            random_seed=42)

print(f"Estatística t: {t:.4f}")
print(f"Valor-p (p-value): {p:.4f}")

Estatística t: 0.7255
Valor-p (p-value): 0.5007


Como p (0.5007) >= 0.05, a diferença no desempenho dos modelos NÃO é estatisticamente significativa.

# Questão 2

In [18]:
import pandas as pd
import numpy as np
from scipy import stats